ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/StateMachine.hs

```diff
@@ -381,6 +387,10 @@ preconditionImpl Model{..} (At (CmdErr cmd err)) =
         case fmap fst . snd <$> M.lookup file (index dbModel) of
           Nothing   -> Bot
           Just bids -> isOpen .&& (thHash $ testHeader tb) `elem` bids
+      PutCorrupted tb        -> isOpen .&& Boolean (and
+        [ (not $ checkBlockIntegrity tb)
+        , (not $ M.member (thHash $ testHeader tb) (mp dbModel))
+        , (slotAfterGC $ Just $ thSlotNo $ testHeader tb) ])
```

Comment:
I would try to use [`Logic`](https://hackage.haskell.org/package/quickcheck-state-machine-0.6.0/docs/Test-StateMachine-Logic.html) as much as possible instead of `Bool`, so you get a more precise message in case it no longer holds. (Also: use `M.notMember` to avoid the extra `Not`)